runtime-transaction/src/runtime_transaction/transaction_view.rs

```diff
@@ -91,6 +100,47 @@ impl<D: TransactionData> RuntimeTransaction<ResolvedTransactionView<D>> {
     }
 }
 
+impl<D: TransactionData> TransactionWithMeta for RuntimeTransaction<ResolvedTransactionView<D>> {
+    fn as_sanitized_transaction(&self) -> impl Borrow<SanitizedTransaction> {
```

Comment:
How about returning a `Cow` instead? Then callers will be more clear about whether they have an owned object or not and don't need to unconditionally clone the `impl Borrow`. As written, this impl allocates a `SanitizedTransaction` and then later we call `as_sanitized_transaction().borrow().clone()` which isn't ideal I think (maybe the compiler is smart enough to not actually clone tho)